avm/res/web/site/certificate/main.bicep

```diff
@@ -0,0 +1,72 @@
+metadata name = 'Web/Function Apps Certificates'
+metadata description = 'This module deploys a Web/Function App Certificate.'
+
+@description('Required. Certificate name.')
+param name string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Optional. Certificate host names.')
+param hostNames array = []
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Kind of resource.')
+param kind string?
+
+@description('Optional. Key Vault resource ID.')
+param keyVaultId string?
+
+@description('Optional. Key Vault secret name.')
+param keyVaultSecretName string?
+
+@description('Optional. Server farm resource ID.')
+param serverFarmResourceId string?
+
+@description('Optional. CNAME of the certificate to be issued via free certificate.')
+param canonicalName string?
+
+@description('Optional. Certificate password.')
+@secure()
+param password string?
+
+@description('Optional. Certificate data in PFX format.')
+@secure()
+param pfxBlob string?
+
+@description('Optional. Method of domain validation for free certificate.')
+param domainValidationMethod string?
+
+resource certificate 'Microsoft.Web/certificates@2024-11-01' = {
+  name: name
+  location: location
+  kind: kind
+  tags: tags
+  properties: {
+    hostNames: !empty(hostNames) ? hostNames : null
+    password: !empty(password) ? password : null
+    pfxBlob: !empty(pfxBlob) ? pfxBlob : null
+    serverFarmId: !empty(serverFarmResourceId) ? serverFarmResourceId : null
+    keyVaultId: !empty(keyVaultId) ? keyVaultId : null
+    keyVaultSecretName: !empty(keyVaultSecretName) ? keyVaultSecretName : null
+    canonicalName: !empty(canonicalName) ? canonicalName : null
+    domainValidationMethod: !empty(domainValidationMethod) ? domainValidationMethod : null
```

Comment:
Woudl strongly recommend to change `hostNames array = []` to `hostNames string[]?` and then update all these lines here to e.g. `pfxBlob: pfxBlob`. If the type is nullable, there's no reason for the condition 😉